internal/client/ca_certificate.go

```diff
@@ -0,0 +1,50 @@
+package infisicalclient
+
+import (
+	"fmt"
+	"terraform-provider-infisical/internal/errors"
+)
+
+const (
+	operationGenerateCACertificate = "CallGenerateCACertificate"
+	operationGetCACertificate      = "CallGetCACertificate"
+)
+
+func (client Client) GenerateCACertificate(request GenerateCACertificateRequest) (GenerateCACertificateResponse, error) {
+	var certResponse GenerateCACertificateResponse
+	response, err := client.Config.HttpClient.
+		R().
+		SetResult(&certResponse).
+		SetHeader("User-Agent", USER_AGENT).
+		SetBody(request).
+		Post(fmt.Sprintf("api/v1/cert-manager/ca/internal/%s/certificate", request.CaId))
+
+	if err != nil {
+		return GenerateCACertificateResponse{}, errors.NewGenericRequestError(operationGenerateCACertificate, err)
+	}
+
+	if response.IsError() {
+		return GenerateCACertificateResponse{}, errors.NewAPIErrorWithResponse(operationGenerateCACertificate, response, nil)
+	}
+
+	return certResponse, nil
+}
+
+func (client Client) GetCACertificate(request GetCACertificateRequest) (GetCACertificateResponse, error) {
+	var certResponse GetCACertificateResponse
+	response, err := client.Config.HttpClient.
+		R().
+		SetResult(&certResponse).
+		SetHeader("User-Agent", USER_AGENT).
+		Get(fmt.Sprintf("api/v1/cert-manager/ca/internal/%s/certificate", request.CaId))
+
+	if err != nil {
+		return GetCACertificateResponse{}, errors.NewGenericRequestError(operationGetCACertificate, err)
+	}
+
```

Comment:
IGNORE PLACEMENT OF COMMENT

Wait why do I get this error when trying to create a CA certificate resource for an intermediate CA? 
<img width="982" height="690" alt="image" src="https://github.com/user-attachments/assets/3bbfb850-1870-49aa-a9b0-45b0f6b8d041" />

I was trying to do the following via TF:
<img width="1512" height="852" alt="image" src="https://github.com/user-attachments/assets/075cbf0e-399a-4de1-82cb-0e4adcb9e831" />